sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/implementation/util/BlobServiceVersionPolicy.java

```diff
@@ -0,0 +1,54 @@
+package com.azure.storage.blob.implementation.util;
+
+import com.azure.core.http.HttpHeaders;
+import com.azure.core.http.HttpPipelineCallContext;
+import com.azure.core.http.HttpPipelineNextPolicy;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import com.azure.storage.blob.BlobServiceVersion;
+import com.azure.storage.common.implementation.Constants;
+import reactor.core.publisher.Mono;
+
+import static com.azure.storage.common.implementation.StorageImplUtils.throwIfContainsHeader;
+
+public class BlobServiceVersionPolicy implements HttpPipelinePolicy {
+
+    @Override
+    public Mono<HttpResponse> process(HttpPipelineCallContext context, HttpPipelineNextPolicy next) {
+        HttpHeaders requestHeaders = context.getHttpRequest().getHeaders();
+        BlobServiceVersion serviceVersion = toServiceVersion(
+            requestHeaders.getValue(Constants.HeaderConstants.SERVICE_VERSION));
+        String ANY_BLOB_API = "any blob API";
```

Comment:
Let's make this a `static final` at the class level, no need to instantiate this each time the policy processes.